lib/internal/url.js

```diff
@@ -1312,7 +1312,9 @@ function domainToUnicode(domain) {
 function urlToOptions(url) {
   var options = {
     protocol: url.protocol,
-    hostname: url.hostname,
+    hostname: url.hostname.startsWith('[') ?
```

Comment:
Nit: from a performance standpoint it would be better to use `charCodeAt`.

Reply:
I wanted to be consistent with https://github.com/nodejs/node/blob/3650972bfb17473b08c5bb32a65adb222355861a/lib/_http_agent.js#L239
Both can be changed in a follow up PR.

Reply:
I don’t think this is a performance-relevant function anyway. The actual HTTP roundtrip will take much longer.